src/publish/mod.rs

```diff
@@ -0,0 +1,194 @@
+//! Types, traits, and functions necessary to publish messages using hedwig
+
+use crate::{Error, Topic, ValidatedMessage};
+use futures_util::{
+    ready,
+    stream::{self, Stream},
+};
+use pin_project::pin_project;
+use std::{
+    collections::BTreeMap,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+pub mod publishers;
```

Comment:
yeah, I can make the publishers and sinks available from the `publish` module without the extra step

Reply:
This only affects the documentation, and here only the `sink` module. I'm fine with still flattening it, but boy it nags me

Reply:
putting `doc(cfg(...))` on the items themselves instead of the module appears to work for all cases. This makes sense, but it multiplies the boilerplate :(